### PR TITLE
Error prone `final` - infrastructure module

### PR DESCRIPTION
**Pull request by gfukushima**
## PR Description
This is PR is part of a series where the aim is to add missing final to enable the `MethodInputParametersMustBeFinal` error prone validation.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java [hunk 1]
```diff
@@ -18,11 +18,16 @@
 import static ethereum.ckzg4844.CKZG4844JNI.BYTES_PER_G2;
 
 import java.util.List;
+import java.util.Objects;
 import org.apache.tuweni.bytes.Bytes;
 
-record TrustedSetup(List<Bytes> g1Points, List<Bytes> g2Points) {
+final class TrustedSetup {
```
Comment (tbenr) [inline infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java]: Why you moved away from `record`?

**Review by tbenr — COMMENTED** [review]
Looks good but I just want to cover the `record` change. We are introducing them in the code base when possible, they make things more concise.

File: infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java (hunk 1 above)
Comment (gfukushima) [reply infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java]: Will have a look at it, but I think this actually my mistake

File: infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java [hunk 2]
```diff
@@ -34,4 +35,9 @@ private void validateG1Point(final Bytes g1Point) {
   private void validateG2Point(final Bytes g2Point) {
     checkArgument(g2Point.size() == BYTES_PER_G2, "Expected G2 point to be %s bytes", BYTES_PER_G2);
   }
+
+  @Override
+  public String toString() {
+    return "TrustedSetup[" + "g1Points=" + g1Points + ", " + "g2Points=" + g2Points + ']';
+  }
```
Comment (tbenr) [inline infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java]: you can remove the toString override too, rolling back to `record`

**Review by tbenr — APPROVED** [review]
LGTM, just a nit on rolling back to `record`

File: infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java (hunk 2 above)
Comment (gfukushima) [reply infrastructure/kzg/src/main/java/tech/pegasys/teku/kzg/TrustedSetup.java]: done, thanks for catching this!

**Pull request merged**